src/components/ui-gauge/ui-gauge.js

```diff
@@ -9,11 +9,8 @@ angular.module('ui').directive('uiGauge', [ '$timeout', '$interpolate',
             link: function(scope, element, attrs) {
                 $timeout(function() {
                     var gauge;
-                    var bgnd = $('#toolbar').css("background-color");
-                    //Detect panel background colour and make foreground text b/w depending.
-                    var colors = $('ui-card-panel').css("background-color").match(/^rgb\((\d+),\s*(\d+),\s*(\d+)\)$/);
-                    var level = ((colors[1]*299) + (colors[2]*587) + (colors[3]*114))/1000;
-                    var fgnd = (level >= 128) ? '#111' : '#eee';
+                    var bgnd = scope.$eval('me.item.waveoptions').waveColor['theme-custom'];
```

Comment:
Why are you setting the bgnd color  to be the waveColor ? - if anything it should be the default colour or whatever the toolbar is... (which may be the wavecolor but may not be at some point) - or is that the only way to get it at this point ?

But yes should no longer need the calculation in here.